lib/scss_lint/linter/property_units.rb

```diff
@@ -0,0 +1,59 @@
+module SCSSLint
+  # Check for allowed units
+  class Linter::PropertyUnits < Linter
+    include LinterRegistry
+
+    def visit_prop(node)
+      @global_allowed = config['global'].to_set
+      @properties = properties config
+      property = node.value.value.to_s
+      name = node_name node
+      units = property.scan /[a-zA-Z%]+/ix
+
+      return if units.empty?
```

Comment:
This gives me an idea for adding support for checking if a property where units are required is lacking units, but that might be harder to pull off. (just thinking out loud)